notifier/notifier.go

```diff
@@ -335,6 +373,24 @@ func (n *Notifier) Alertmanagers() []*url.URL {
 	return res
 }
 
+// AlertmanagerCount returns the number of Alertmangers currently
+// active from the Alertmanager discovery.
+func (n *Notifier) AlertmanagerCount() int {
```

Comment:
Do we really need a new method here with complex locking logic. Just using `len(n.Alertmanagers())` should be fine, no?